requests/utils.py

```diff
@@ -447,6 +446,36 @@ def get_encodings_from_content(content):
             xml_re.findall(content))
 
 
+def _parse_content_type_header(header):
+    """Returns content type and parameters from given header
+
+    :param header: string
+    :return: tuple containing content type and dictionary of
+         parameters
+    """
+    if not header:
+        return None
+    # append delimiter on end to ensure at least two elements when split by ';'
+    header += ';'
+    # split content type's main value from params
+    tokens = header.split(';', 1)
+    content_type_index = 0
+    params_index = 1
+
+    content_type = tokens[content_type_index].strip()
+    params = tokens[params_index]
+    params_dict = dict()
+
+    for param in params.split(';'):
```

Comment:
The above said, if we weren't restricting it to split once we could simplify the code further and do:

```
tokens = header.split(';')
content_type, params = tokens[0].strip(), tokens[1:]
params_dict = {}. # Using dict is actually slower than a dictionary literal. Weird but tru
for param in params:
    # ... What you have below

return content_type, params_dict
```